lib/mapObjects/MiscObjects.cpp

```diff
@@ -490,7 +490,7 @@ void CGMonolith::onHeroVisit( const CGHeroInstance * h ) const
 	TeleportDialog td(h->id, channel);
 	if(isEntrance())
 	{
-		if(cb->isTeleportChannelBidirectional(channel) && 1 < cb->getTeleportChannelExits(channel).size())
+		if(cb->getTeleportChannelExits(channel).size() > 1)
```

Comment:
Unlikely we should change server code to allow AI going through one-way monoliths. Do we have monolith in AI path? If not you may want to make changes similar to Whirlpool. Most likely the one in AIPathNode::commit which sets theNodeBefore

Reply:
This is more of a global and not only AI related issue. I saw the error message (screenshot in the top) some time ago already, while playing. oneWayTeleports (UniDirectional) can have multiple exits with one entrance. So the check with isTeleportChannelBidirectional failed and exit was taken randomly. Then client also used RNG and sometimes it hit the right one, and sometimes not. Am I understanding it correctly or the problem is somewhere else ?

Reply:
Strange, I still do not understand how it works for you. First of all these settings should be false by default I beleave

> , useTeleportOneWay(VLC->settings()->getBoolean(EGameSettings::PATHFINDER_USE_MONOLITH_ONE_WAY_UNIQUE))
> , useTeleportOneWayRandom(VLC->settings()->getBoolean(EGameSettings::PATHFINDER_USE_MONOLITH_ONE_WAY_RANDOM))

Reply:
I'm totally surprised... Have to look into your commits because I reverted changes to initial code in terms of the above Bidirectional check and **it works without errors now**! Probably should've check before committing